cfpb/consumerfinance.gov
# Stub GovDelivery API in Cypress tests

**Fixes GHE/CFGOV/platform/issues/4480**

Our [Cypress tests for the email signup module](https://github.com/cfpb/consumerfinance.gov/blob/main/test/cypress/integration/components/email-signup/email-signup.cy.js) check to make sure that when a valid email address is submitted, a success message appears as expected. Right now, that success message is generated by [hitting our subscription API via the email signup form and receiving a success response](https://github.com/cfpb/consumerfinance.gov/blob/main/test/cypress/integration/components/email-signup/email-signup-helpers.cy.js). That works fine in a local environment, where we [mock GovDelivery API calls on the back end and always return a success response](https://github.com/cfpb/consumerfinance.gov/blob/main/.env_SAMPLE#L115). When we run Cypress tests as part of our EKS deployment pipeline, though, the tests run in a production environment and are thus trying (and failing) to hit the GovDelivery API. Questions of whether an EKS deployment should be able to hit the GovDelivery API aside, we should be intercepting any API requests in our Cypress tests so that we're not inadvertently testing whether an API works when we're trying to test functionality. We do a similar [intercept for Census API requests in the rural and underserved tool tests](https://github.com/cfpb/consumerfinance.gov/blob/main/test/cypress/integration/consumer-tools/rural-or-underserved-tool/rural-or-underserved-tool-helpers.cy.js#L23-L57), and we want to apply that same intercept pattern in the email signup tests.

---

## Additions

- Intercept requests to `/subscriptions/new/` in Cypress tests

## How to test this PR

1. Run the email signup Cypress tests with `yarn cypress open` and selecting the `email-signup.cy.js` spec
2. Note the `(xhr) POST 200 /subscriptions/new/` in step 7 shows that the response was stubbed (see screenshot below)
3. If you want to be extra thorough, try changing the fixture in `subscription.json` to be `"result": "fail"`. That should make the failure message appear in the email signup module instead of the success message, which will make the email signup tests fail  (and prove the response is stubbed from the fixture and not coming from the subscriptions API).

## Screenshots

Before | After
--- | ---
![before](https://user-images.githubusercontent.com/1862695/193690848-1cf1aab6-4641-495d-aa0f-45cada66d364.png) | ![after](https://user-images.githubusercontent.com/1862695/193690889-9180f5f5-d712-4a8a-b9c1-6094c09d9a28.png)

## Checklist

  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
